lib/Service/ParticipantService.php

```diff
@@ -426,9 +433,25 @@ public function addUsers(Room $room, array $participants, ?IUser $addedBy = null
 			$this->dispatcher->dispatchTyped($attendeeEvent);
 
 			$this->dispatcher->dispatch(Room::EVENT_AFTER_USERS_ADD, $event);
+
+			$lastMessage = $event->getLastMessage();
+			if ($lastMessage instanceof IComment) {
+				$this->updateRoomLastMessage($room, $lastMessage);
+			}
 		}
 	}
 
+	protected function updateRoomLastMessage(Room $room, IComment $message): void {
+		$room->setLastMessage($message);
+		$lastMessageCache = $this->cacheFactory->createDistributed('talk/lastmsgid');
```

Comment:
Don't will me more readable?
```suggestion
		$lastMessageCache = $this->cacheFactory->createDistributed('talk/last_message_id');
```

Reply:
No, this is already in use in other places and we need to purge the correct one